Article: Fleet 4.50.0 release (#19211)
fleetdm/fleet: articles/fleet-4.50.0.md

For upgrade instructions, see our [upgrade guide](https://fleetdm.com/docs/deploying/upgrading-fleet) in the Fleet docs.

## Highlights

* Deploy security agents to macOS, Windows, and Linux
* Policy description and resolutions aided by AI
* Mac Admins SOFA support
* `zsh` support

## Deploy security agents to macOS, Windows, and Linux

Fleet enhances the deployment capabilities for IT administrators, particularly concerning security agents. Now available in Fleet Premium, this feature allows administrators to add and deploy security agents directly to macOS, Windows, and Linux hosts through the Software page, the Fleet API, or via GitOps workflows. This deployment functionality requires that the host has a `fleetd` agent with scripts enabled, but notably, it does not necessitate MDM (Mobile Device Management) features to be enabled within Fleet. This new capability supports a more streamlined and efficient approach to enhancing host security across diverse operating environments, allowing IT and security teams to ensure their hosts are protected with the necessary security tools without the complexity of additional infrastructure changes.

> 

Reviewer: ```suggestion

For users that host Fleet themselves, this feature requires hooking up Fleet with an S3 bucket. See how in the server configuration reference [here](https://fleetdm.com/docs/configuration/fleet-server-configuration#s-3).
```

Suggest adding a not about the new infra requirement: S3

Note that the doc link is currently broken however it will be updated as part of this PR: https://github.com/fleetdm/fleet/pull/19236

## `zsh` support

Fleet has expanded its scripting capabilities by adding support for `zsh` (Z Shell) scripts, catering to IT administrators' and developers' diverse scripting preferences. This update allows users to execute `zsh` scripts directly within Fleet, providing a flexible and powerful toolset for managing and automating tasks across various systems. By accommodating `zsh`, known for its robust features and interactive use enhancements over `bash`, Fleet enhances its utility for more sophisticated script operations. This support not only broadens the scope of administrative scripts that can be run but also aligns with the ongoing efforts to adapt to the evolving needs of users in dynamic IT environments.

## Changes

> ### Endpoint Operations

Reviewer: ```suggestion
### Endpoint operations
```

[Sentence case](https://fleetdm.com/handbook/company/why-this-way#why-does-fleet-use-sentence-case)

- Added optional AI-generated policy descriptions and remediations. 
- Added flag to enable deletion of old activities and associated data in cleanup cron job.
- Added support for escaping `$` (with `\`) in gitops yaml files.
- Optimized policy_stats updates to not lock the policy_membership table.
- Optimized the hourly host_software count query to reduce individual query runtime.
- Updated built-in labels to support being applied via `fleetctl apply`.

> ### Device Management (MDM)

Reviewer: ```suggestion
### Device management (MDM)
```

- Added functionality to filter hosts by software installer status.
- Added support to the global activity feed for "Added software" and "Deleted software" actions.
- Added the `POST /api/fleet/orbit/software_install/result` endpoint for fleetd to send results for a software installation attempt.
- Added the `GET /api/v1/fleet/hosts/{id}/software` endpoint to list the installed software for the host.
- Added support for uploading and running zsh scripts on macOS and Linux hosts.
- Added the `cron` job to periodically remove unused software installers from the store.
- Added a new command `fleetctl api` to easily use fleetctl to hit any REST endpoint via the CLI.
- Added support to extract package name and version from software installers.
- Added the uninstalled but available software installers to the response payload of the "List software titles" endpoint.
- Updated MySQL host_operating_system insert statement to reduce table lock time.
- Updated software page to support new add software feature.
- Updated fleetctl to print team id as part of the `fleetctl get teams` command.
- Implemented an S3-based and local filesystem-based storage abstraction for software installers.

> ### Vulnerability Management

Reviewer: ```suggestion
### Vulnerability management
```

This improvement enhances the user experience for administrators and end-users by enabling transparent communication of policy purposes and actions to end-users. This can be especially useful in scenarios like scheduled [maintenance windows](https://fleetdm.com/announcements/fleet-in-your-calendar-introducing-maintenance-windows) visible to users through calendar events or device notifications. By automating the generation of detailed, relevant policy descriptions, Fleet helps ensure that all parties understand what each policy entails and why it is important, enhancing the organization's overall security posture and compliance.

## Mac Admins SOFA support

> Fleet has integrated support for the Mac Admins [SOFA](https://github.com/macadmins/sofa) (Structured Open Feed Aggregator), enhancing its capabilities to provide comprehensive tracking and surfacing of update information for macOS and iOS environments. SOFA, known for its machine-readable feed and user-friendly web interface, offers continuous updates on XProtect data, OS updates, and detailed release information. This integration within Fleet is facilitated through new tables specifically for security release information ([`sofa_security_release_info`](https://fleetdm.com/tables/sofa_security_release_info)) and unpatched CVEs ([`sofa_unpatched_cves`](https://fleetdm.com/tables/sofa_unpatched_cves)).

Reviewer 2: ```suggestion
Fleet has integrated support for the Mac Admins [SOFA](https://github.com/macadmins/sofa) (Structured Open Feed Aggregator), enhancing its capabilities to provide comprehensive tracking and surfacing of update information for macOS hosts. SOFA, known for its machine-readable feed and user-friendly web interface, offers continuous updates on XProtect data, OS updates, and detailed release information. This integration within Fleet is facilitated through new tables specifically for security release information ([`sofa_security_release_info`](https://fleetdm.com/tables/sofa_security_release_info)) and unpatched CVEs ([`sofa_unpatched_cves`](https://fleetdm.com/tables/sofa_unpatched_cves)).
```

> Fleet has integrated support for the Mac Admins [SOFA](https://github.com/macadmins/sofa) (Structured Open Feed Aggregator), enhancing its capabilities to provide comprehensive tracking and surfacing of update information for macOS and iOS environments. SOFA, known for its machine-readable feed and user-friendly web interface, offers continuous updates on XProtect data, OS updates, and detailed release information. This integration within Fleet is facilitated through the recent updates to the [Mac Admins osquery extension](https://github.com/macadmins/osquery-extension), which now includes tables specifically for security release information (`sofa_security_release_info`) and unpatched CVEs (`sofa_unpatched_cves`).

Reviewer: ```suggestion
Fleet has integrated support for the Mac Admins [SOFA](https://github.com/macadmins/sofa) (Structured Open Feed Aggregator), enhancing its capabilities to provide comprehensive tracking and surfacing of update information for macOS and iOS environments. SOFA, known for its machine-readable feed and user-friendly web interface, offers continuous updates on XProtect data, OS updates, and detailed release information. This integration within Fleet is facilitated through new tables specifically for security release information ([`sofa_security_release_info`](https://fleetdm.com/tables/sofa_security_release_info)) and unpatched CVEs ([`sofa_unpatched_cves`](https://fleetdm.com/tables/sofa_unpatched_cves)).
```

Adding links to the tables on fleetdm.com/tables. 

Suggest prioritizing that over MacAdmins osquery extension. We're bundling these tables in the fleetd agent. Folks don't need to worry about the MacAdmins extension.

> Fleet has integrated support for the Mac Admins [SOFA](https://github.com/macadmins/sofa) (Structured Open Feed Aggregator), enhancing its capabilities to provide comprehensive tracking and surfacing of update information for macOS and iOS environments. SOFA, known for its machine-readable feed and user-friendly web interface, offers continuous updates on XProtect data, OS updates, and detailed release information. This integration within Fleet is facilitated through the recent updates to the [Mac Admins osquery extension](https://github.com/macadmins/osquery-extension), which now includes tables specifically for security release information (`sofa_security_release_info`) and unpatched CVEs (`sofa_unpatched_cves`).

Reviewer: ```suggestion
Fleet has integrated support for the Mac Admins [SOFA](https://github.com/macadmins/sofa) (Structured Open Feed Aggregator), enhancing its capabilities to provide comprehensive tracking and surfacing of update information for macOS hosts. SOFA, known for its machine-readable feed and user-friendly web interface, offers continuous updates on XProtect data, OS updates, and detailed release information. This integration within Fleet is facilitated through the recent updates to the [Mac Admins osquery extension](https://github.com/macadmins/osquery-extension), which now includes tables specifically for security release information (`sofa_security_release_info`) and unpatched CVEs (`sofa_unpatched_cves`).
```

No iOS in Fleet (yet) ;)

Reviewer 2: It won't let me commit this because it's outdated. I'll put in another suggestion.
